### DISCUSSION TITLE
Group Classes (breakpoints, hover, group etc.)

### DISCUSSION THREAD

**Discussion by davidhellmann** [tailwindlabs/tailwindcss#2996] [Ideas]
Hey, 

dived more into tailwind this week. I like the approach more and more but I don't like the structure.
Some little thoughts how it is better structured and more readable

## Current State
One liner
```html
<div class="w-screen px-2 bg-red-100 hover:bg-blue-100 md:w-6/12 md:mx-auto md:px-4 md:bg-red-200 md:hover:bg-blue-200 lg:bg-red-400 lg:hover:bg-blue-400"></div>
```

With line breaks
```html
<div class="w-screen px-2 bg-red-100
            hover:bg-blue-100
            md:w-6/12 md:mx-auto md:px-4 md:bg-red-200 
            md:hover:bg-blue-200 
            lg:bg-red-400  
            lg:hover:bg-blue-400"></div>
```

## Suggestion
One liner
```html
<div class="w-screen px-2 bg-red-100 hover:(bg-blue-100) md:(w-6/12 mx-auto px-4 bg-red-200) md:hover:(bg-blue-100) lg:(bg-red-400) lg:hover:(bg-blue-400)"></div>
```

With line breaks
```html
<div class="w-screen px-2 bg-red-100
            hover:(bg-blue-100)
            md:(w-6/12 mx-auto px-4 bg-red-200) 
            md:hover:(bg-blue-100)
            lg:(bg-red-400)
            lg:hover:(bg-blue-400)"></div>
```

## What problem does it solve?
Group classes and make code scanning and reading easier. Easier copy / paste / delete class fragments. And sure, less DRY code.

**Comment by jmartsch** [comment]
It is easier to read.
But with this approach it would be difficult, to add a class for a certain breakpoint using JavaScript.

**Comment by davidhellmann** [comment]
@jmartsch valid! But maybe this patterns can coexist? For me this are two use cases. 
First I write classes by myself and here it's important for me that I can scan all this stuff smooth as possible and I think groups can help a lot here. Maybe it would be possible to collapse groups like this in the editor with a shortcut (Collapse groups).

**Comment by boompttn** [comment]
Something similar https://github.com/Noriller/easy-tailwind

**Comment by yf-hk** [comment]
Try twin.macro, it does support this kind of grouping

**Comment by nistadev** [comment]
I totally agree would be a great approach, I even created a [discussion](https://github.com/tailwindlabs/tailwindcss/discussions/11472) with the exact same approach before finding yours!
